core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -289,7 +346,95 @@ public static HighlightBuilder fromXContent(QueryParseContext parseContext) thro
         return highlightBuilder;
     }
 
+    public void parse(XContentParser parser, SearchContext context) throws Exception {
```

Comment:
Sure, I thought HighlightBuilder should eventually implement SearchParseElement, but its not needed right now. Will make the build() method which produces the SearchContextHighlight public then.